components/hab/src/command/pkg/download.rs

```diff
@@ -245,9 +254,10 @@ impl<'a> DownloadTask<'a> {
 
         // We rely on the fact that all packages are always added to the unstable channel as well as
         // any other channel specified.
-        let mut channel = &ChannelIdent::unstable();
-        if !ident.fully_qualified() {
-            channel = self.channel
+        let channel = if ident.fully_qualified() {
+            ChannelIdent::unstable()
+        } else {
+            (*channel).clone()
```

Comment:
I wonder if we can avoid doing a clone..

Reply:
Fixed